### PR TITLE
Add Ability to Set Enum Properties to TempoWorld

### PR DESCRIPTION
**Pull request by pete-tempo**
Adds the ability to set Enum properties (as FEnumProperty or FByteProperty) on Actors and Components via RPC to TempoWorld.

Confirmed working with these three common ways to make an Enum in UE:
```
UENUM(BlueprintType)
enum ETestEnumA
{
	EAOne = 0,
	EATwo = 1,
	EAThree = 2
};

UENUM(BlueprintType)
enum class ETestEnumB : uint8
{
	EBOne = 0,
	EBTwo = 1,
	EBThree = 2
};

UENUM(BlueprintType)
namespace ETestEnumC
{
	enum Type : int32
	{
		ECOne = 0,
		ECTwo = 1,
		ECThree = 2
	};
}

UCLASS()
class MYGAME_API AEnumTestActor : public AActor
{
	GENERATED_BODY()

public:
	UPROPERTY(EditAnywhere)
	TEnumAsByte<ETestEnumA> EnumA;

	UPROPERTY(EditAnywhere)
	ETestEnumB EnumB;

	UPROPERTY(EditAnywhere)
	TEnumAsByte<ETestEnumC::Type> EnumC;
};
```

### CODE REVIEW

File: TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp [hunk 1]
```diff
@@ -734,6 +736,11 @@ grpc::Status GetPropertyForRequest(const UObject* Object, const RequestType& Req
 	const UClass* Class = Object->GetClass();
 	Property = Class->FindPropertyByName(FName(PropertyName));
 
+	if (!Property)
+	{
+		return grpc::Status(grpc::FAILED_PRECONDITION, "Property not found");
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp:741]: This is an unrelated fix

File: TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp [hunk 2]
```diff
@@ -945,7 +1003,8 @@ grpc::Status SetPropertyImpl<SetFloatPropertyRequest>(const UWorld* World, const
 {
 	// First try to set it as a double, then fall back on float
 	const grpc::Status FloatStatus = SetSinglePropertyImpl<FDoubleProperty>(World, Request, Request.value());
-	if (FloatStatus.ok())
+	// If we got an error other than FAILED_PRECONDITION that means the type was right, but something else was wrong.
+	if (FloatStatus.ok() || FloatStatus.error_code() != grpc::FAILED_PRECONDITION)
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp:1007]: This is an unrelated fix. If the property was right but we got another error, we should quit and report that error.

File: TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp [hunk 3]
```diff
@@ -958,14 +1017,29 @@ grpc::Status SetPropertyImpl<SetStringPropertyRequest>(const UWorld* World, cons
...
 		return StrStatus;
 	}
 	const FName ValueName(UTF8_TO_TCHAR(Request.value().c_str()));
 	return SetSinglePropertyImpl<FNameProperty>(World, Request, ValueName);
 }
 
+template<>
+grpc::Status SetPropertyImpl<SetEnumPropertyRequest>(const UWorld* World, const SetEnumPropertyRequest& Request)
+{
+	// First try to set it as an FEnumProperty, then fall back on FByteProperty, then fall back on FIntProperty
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp:1032]: Not FIntProperty

File: TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp [hunk 4]
```diff
@@ -742,6 +749,58 @@ grpc::Status GetPropertyForRequest(const UObject* Object, const RequestType& Req
...
+	}
+	*static_cast<int64*>(ValuePtr) = Value;
+	return grpc::Status_OK;
+}
+
+template <>
+grpc::Status SetSinglePropertyValue<FByteProperty, FString>(void* ValuePtr, FByteProperty* Property, const FString& ValueStr)
+{
+	const UEnum* PropertyEnum = Property->Enum;
+	const int32 Value = GetEnumValueByAuthoredName(PropertyEnum, ValueStr);
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoWorld/Private/TempoActorControlServiceSubsystem.cpp:795]: int64?

**Review by gio-tempo — APPROVED** [review]
Reviewed over VC.  Acknowledge coming changes.
